src/solver/conv_ck_igemm_fwd_bias_activ.cpp

```diff
@@ -0,0 +1,421 @@
+        const_cast<void*>( // NOLINT (cppcoreguidelines-pro-type-const-cast)
+            static_cast<const void*>(invoke_ctx.in)),
+        const_cast<void*>( // NOLINT (cppcoreguidelines-pro-type-const-cast)
+            static_cast<const void*>(wei_buf)),
+        invoke_ctx.out,
+        const_cast<void*>(static_cast<const void*>(bias_buf)),
+        args.N,
+        args.K,
+        args.C,
+        args.input,
+        args.filter,
+        args.output,
+        args.strides,
+        args.dilation,
+        args.lPadding,
+        args.rPadding,
+        {},
+        {},
+        {});
+    auto invoker_ptr            = conv_ptr->MakeInvokerPointer();
+    const auto enable_profiling = handle.IsProfilingEnabled();
+
+    float elapsed_time =
+        invoker_ptr->Run(argument_ptr.get(), {handle.GetStream(), enable_profiling});
+    if(enable_profiling)
+    {
+        handle.ResetKernelTime();
+        handle.AccumKernelTime(elapsed_time);
+    }
+}
+#endif
+
+void PerformanceConfigConvCKIgemmFwdBiasActiv::HeuristicInit(const FusionContext& ctx)
+{
+#if !MIOPEN_BACKEND_HIP || !MIOPEN_USE_COMPOSABLEKERNEL
+    std::ignore = ctx;
+#else
+    const auto& conv_prob = ctx.problem.GetConvProblem(0, conv::Direction::Forward).conv_problem;
+    switch(conv_prob.GetInDataType())
+    {
+    case miopenInt8:
+    case miopenHalf: Init<ck::half_t>(conv_prob); break;
+    case miopenFloat:
+    case miopenInt32:
+    case miopenInt8x4:
+    case miopenBFloat16:
+    case miopenDouble: break;
+    }
+#endif
+}
+
+bool PerformanceConfigConvCKIgemmFwdBiasActiv::SetNextValue(const FusionContext& ctx)
+{
+    if(this->total_size == -1)
+    {
+        this->HeuristicInit(ctx);
+        assert(this->total_size != -1);
+        return true;
+    }
+    if((this->index + 1) < this->total_size)
+    {
+        ++this->index;
+        const auto conv = GetInstances();
+        this->kernel_id = conv[this->index]->GetTypeString();
+        return true;
+    }
+    return false;
+}
+
+bool PerformanceConfigConvCKIgemmFwdBiasActiv::IsValidValue() const
+{
+    return this->index < this->total_size;
+}
+
+bool PerformanceConfigConvCKIgemmFwdBiasActiv::IsValid(const FusionContext& ctx) const
+{
+#if !MIOPEN_BACKEND_HIP || !MIOPEN_USE_COMPOSABLEKERNEL
+    std::ignore = ctx;
+    return false;
+#else
+    // Extract convolution problem from the fusion context.
+    const auto& problem = ctx.problem.GetConvProblem(0, conv::Direction::Forward);
+    switch(problem.conv_problem.GetInDataType())
+    {
+    case miopenInt8:
+    case miopenHalf: return CheckIsSupportCKArgs<ck::half_t>(problem);
+    case miopenFloat:
+    case miopenInt32:
+    case miopenInt8x4:
+    case miopenBFloat16:
+    case miopenDouble: break;
+    }
+    return false;
+#endif
+}
+
+bool PerformanceConfigConvCKIgemmFwdBiasActiv::operator==(
+    const PerformanceConfigConvCKIgemmFwdBiasActiv& other) const
+{
+    return this->index == other.index && this->total_size == other.total_size;
```

Comment:
total_size is either initialized to "unknown" (-1) or dynamically computed in Init(). How it could be so that two instances have different total_size? Can you please provide an example scenario that leads to this mismatch?

Maybe we should throw in case of mismatch?

Reply:
ended up just comparing kernel_id (name)